Add ph-no-capture to relevant sections

yakkomajuri opened the pull request:
## Changes

*Please describe.*  
*If this affects the frontend, include screenshots.*  

Discussion should go here: https://github.com/PostHog/ops/issues/166

Minor fixes for a few TS issues I found along the way.

## Checklist

- [ ] All querysets/queries filter by Organization, by Team, and by User
- [ ] Django backend tests
- [ ] Jest frontend tests
- [ ] Cypress end-to-end tests

yakkomajuri committed:
properly configure ph-no-capture

yakkomajuri commented:
Code quality failing as a result of other TS errors on the files I edited.

paolodamico commented:
Added some comments on the related ops issue which would love to address before actually reviewing the code of this PR.

yakkomajuri committed:
better precision

yakkomajuri commented:
@paolodamico please take a look. Better precision here.

Also got rid of a bunch of TS errors on annotations, although for great typing there the logic would need to be typed, and that's beyond the scope here.

yakkomajuri commented on frontend/src/scenes/events/EventsVolumeTable.tsx:
```diff
@@ -26,7 +26,8 @@ export function EventsVolumeTable(): JSX.Element {
             },
-            render: (item: EventUsageType) => humanizeNumber(item.volume),
+            // eslint-disable-next-line react/display-name
+            render: (item: EventUsageType) => <span className="ph-no-capture">{humanizeNumber(item.volume)}</span>,
```
context: https://stackoverflow.com/questions/55620562/eslint-component-definition-is-missing-displayname-react-display-name

yakkomajuri committed:
Merge branch 'master' into sr-privacy

paolodamico commented on frontend/src/lib/components/PropertiesTable.tsx:
```diff
@@ -104,7 +104,7 @@ function ValueDisplay({ value, rootKey, onEdit, nestingLevel }: ValueDisplayType
 
     return (
-        <div className="properties-table-value">
+        <div className="properties-table-value ph-no-capture">
```
I think we can be more precise here and only sanitize the actual value as opposed to the type of value indicator, extra link, etc.

paolodamico commented on frontend/src/lib/components/PropertiesTable.tsx:
```diff
@@ -170,7 +170,7 @@ export function PropertiesTable({
             render: function Key(item: any): JSX.Element {
                 return (
-                    <div className="properties-table-key">
+                    <div className="properties-table-key ph-no-capture">
```
Question about properties name on main issue.

paolodamico committed:
prettier & few tweaks

paolodamico committed:
use already existing type

paolodamico commented on frontend/src/scenes/experimentation/FeatureFlags.js:
```diff
@@ -122,7 +122,7 @@ function _FeatureFlags() {
                 })}
                 size="small"
-                rowClassName={'cursor-pointer ' + rrwebBlockClass}
+                rowClassName={'cursor-pointer ' + 'ph-no-capture'}
```
I know you're not changing anything here, but should feature flag names really be obscured?

yakkomajuri replied:
Oh this I feel more strongly about than prop names. A feature flag name gives me insight into the experiments a company is running. Happy to expand more on the main issue.

paolodamico replied:
Alright makes sense, in that case I only think it would be useful to black out specifically the feature flag name and keep everything else?

paolodamico commented on frontend/src/scenes/insights/Insights.js:
```diff
@@ -213,6 +213,7 @@ function _Insights() {
                         They each have generic logic that updates the url
                         */}
                                 <Card
+                                    className="ph-no-capture"
```
I think there's a bunch of stuff here that doesn't need to be obscured (e.g. filters, error/timeout messages, filters tabs, ...)

yakkomajuri replied:
oh maybe this lingered - thought I had updated. filters I think should be.

paolodamico replied:
By filters I meant like weekly/daily/monthly, type of graph, ...., not actual filters, sorry for the confusion. I believe filters we do obscure very targeted.

paolodamico commented on frontend/src/scenes/me/Settings/index.tsx:
```diff
@@ -20,15 +20,17 @@ function _MySettings(): JSX.Element {
-                    Personal API Keys
-                </h2>
-                <PersonalAPIKeys />
+                <span className="ph-no-capture">
```
This one I'm less opposed to, but I do think we could sanitize the API keys specifically & the password input instead of the whole thing.

paolodamico commented on frontend/src/scenes/persons/Cohorts.tsx:
```diff
@@ -154,7 +154,7 @@ function _Cohorts(): JSX.Element {
                     rowKey="id"
                     pagination={{ pageSize: 100, hideOnSinglePage: true }}
-                    rowClassName={'cursor-pointer ' + rrwebBlockClass}
+                    rowClassName={'cursor-pointer ' + 'ph-no-capture'}
```
Same as feature flags, think the only thing worth obscuring is the number of users in cohort, and at most the cohort name.

paolodamico commented on frontend/src/scenes/sessions/SessionsView.tsx:
```diff
@@ -213,6 +210,7 @@ export function SessionsView({ personIds, isPersonPage = false }: SessionsTableP
             </div>
 
             <Table
+                className="ph-no-capture"
```
Don't think sanitizing the whole table is necessary, how about the Person & Event columns?

paolodamico commented on frontend/src/scenes/events/PropertiesVolumeTable.tsx:
```diff
@@ -69,17 +71,19 @@ export function PropertiesVolumeTable(): JSX.Element {
             )}
             <Table
-                dataSource={user?.team.event_properties_with_usage
+                className="ph-no-capture"
```
Added a comment in the main issue about whether to capture property names. Regardless, I think it's worth sanitizing each column instead of the whole table, for recordings to be more useful.

paolodamico commented on frontend/src/scenes/experimentation/EditFeatureFlag.js:
```diff
@@ -146,7 +146,7 @@ export function EditFeatureFlag({ featureFlag, logic, isNew }) {
 
                     <Form.Item
-                        className={rrwebBlockClass}
+                        className="ph-no-capture"
```
I think we already sanitize more precisely on `PropertyFilters`?

paolodamico commented on frontend/src/scenes/sessions/SessionsPlay.tsx:
```diff
@@ -96,7 +96,7 @@ function _SessionsPlay(): JSX.Element {
 
     return (
-        <div className="session-player">
+        <div className="session-player ph-no-capture">
```
Not sure everything is sensitive, think we can keep stuff like the timeline, the device/resolution indicators, loaders, ...

paolodamico committed:
minor tweaks

paolodamico reviewed (changes requested):
Overall I think this is pretty great! Reviewed inline (hard to test everything), feel free to push back if you think it makes sense.

As a side note, I think it'll make sense to delete older recordings (let's just wait some time)  to make sure we don't accidentally keep any information in there that we shouldn't.

Twixes commented:
Do we want to go forward with this?

yakkomajuri commented:
Yup - I've been a bit out of time. Think this is important. Will address @paolodamico's feedback early next week and get this in.

yakkomajuri committed:
merge master

yakkomajuri committed:
much better precision

yakkomajuri committed:
allow prop names

yakkomajuri commented:
Addressed all your feedback @paolodamico - no large areas are blocked anymore, and I've allowed things like prop names.

Would love to see this in as I believe it is very important.

yakkomajuri commented:
I guess the merge wasn't clean - will look into it

yakkomajuri commented:
Closing in favor of https://github.com/PostHog/posthog/pull/3596
